journey-map/public/render.mjs

```diff
@@ -70,73 +26,69 @@ const expandParents = (journeyStates, otherStates) => {
 
-                Object.entries(expandedDefinition.events || {}).forEach(([evt, def]) => {
-                    // Map target states to expanded states
-                    if (def.targetState && !def.targetJourney) {
-                        def.targetState = `${def.targetState}_${state}`;
-                    }
+                Object.entries(expandedDefinition.events || {}).forEach(([evt, eventDef]) => {
+                    mapTargetStateToExpandedState(eventDef, subJourneyState);
 
                     // Map exit events to targets in the parent definition
-                    const exitEvent = def.exitEventToEmit;
+                    const exitEvent = eventDef.exitEventToEmit;
                     if (exitEvent) {
-                        delete def.exitEventToEmit;
+                        delete eventDef.exitEventToEmit;
                         if (definition.exitEvents[exitEvent]) {
-                            Object.assign(def, definition.exitEvents[exitEvent]);
+                            Object.assign(eventDef, definition.exitEvents[exitEvent]);
                         } else {
-                            console.warn(`Unhandled exit event from ${state}:`, exitEvent)
+                            console.warn(`Unhandled exit event from ${subJourneyState}:`, exitEvent)
                             delete expandedDefinition.events[evt];
                         }
                     }
                 });
 
-                journeyMap[`${nestedState}_${state}`] = expandedDefinition;
+                journeyMap[`${nestedState}_${subJourneyState}`] = expandedDefinition;
             });
 
+            // Make a copy of the event definition to avoid mutating the original
+            const entryEvents = JSON.parse(JSON.stringify(subjourney.entryEvents));
             // Update entry events on other states to expanded states
-            Object.entries(subjourney.entryEvents).forEach(([entryEvent, def]) => {
+            Object.entries(entryEvents).forEach(([entryEvent, entryEventDef]) => {
+                mapTargetStateToExpandedState(entryEventDef, subJourneyState);
+
                 Object.values(journeyMap).forEach((journeyDef) => {
                     if (journeyDef.events?.[entryEvent]) {
-                        const target = resolveEventTarget(journeyDef.events[entryEvent], formData);
-                        if (target.targetState === state && !target.targetEntryEvent) {
-                            target.targetState = `${def.targetState}_${state}`;
+                        const target = resolveEventTargets(journeyDef.events[entryEvent], undefined, formData).find(t => !t.journeyContext)
+                        if (target.targetState === subJourneyState && !target.targetEntryEvent) {// if (target.targetState === subJourneyState && !target.targetEntryEvent) {
+                            journeyDef.events[entryEvent] = entryEventDef;
```

Comment:
We replace the event with the whole target definition as we also want to include the `{..., checkJourneyContext}` bits which is handled in `renderTransitions`